src/Microsoft.ML.HalLearners/SymSgdClassificationTrainer.cs

```diff
@@ -26,16 +25,14 @@
     SymSgdClassificationTrainer.LoadNameValue,
     SymSgdClassificationTrainer.ShortName)]
 
-[assembly: LoadableClass(typeof(void), typeof(SymSgdClassificationTrainer), null, typeof(SignatureEntryPointModule), "SymSGD")]
+[assembly: LoadableClass(typeof(void), typeof(SymSgdClassificationTrainer), null, typeof(SignatureEntryPointModule), SymSgdClassificationTrainer.LoadNameValue)]
 
 namespace Microsoft.ML.Runtime.SymSgd
 {
-    using TPredictor = IPredictorWithFeatureWeights<Float>;
+    using TPredictor = IPredictorWithFeatureWeights<float>;
 
     /// <include file='doc.xml' path='doc/members/member[@name="SymSGD"]/*' />
-    public sealed class SymSgdClassificationTrainer :
-        TrainerBase<TPredictor>,
-        ITrainer<TPredictor>
+    public sealed class SymSgdClassificationTrainer : TrainerEstimatorBase<BinaryPredictionTransformer<TPredictor>, TPredictor>
```

Comment:
>TrainerEstimatorBase [](start = 54, length = 20)

SingleFeatureTrainerEstimatorBase

Reply:
Sorry I can't find that. what is the name of the base class I 
should derive from?

---
In reply to: [220056284](https://github.com/dotnet/machinelearning/pull/1012#discussion_r220056284) [](ancestors = 220056284)

Reply:
Why would this not work by the way?


---
In reply to: [220256710](https://github.com/dotnet/machinelearning/pull/1012#discussion_r220256710) [](ancestors = 220256710,220056284)

Reply:
My bad, this is fine. got it confused with the TransformerPredictors 

---
In reply to: [220256891](https://github.com/dotnet/machinelearning/pull/1012#discussion_r220256891) [](ancestors = 220256891,220256710,220056284)